spec/example_app/app/controllers/docs_controller.rb

```diff
@@ -9,7 +16,15 @@ def index
   end
 
   def show
-    render_page "docs/#{params[:page]}"
+    if check_special_file.length > 0 
+      render_page check_special_file[0][:file]
+    else
+      render_page "docs/#{params[:page]}"
+    end
+  end
+
+  def check_special_file
+    SPECIAL_FILES.select { |page| page[:page] == params[:page] }
```

Comment:
If we were to use [`first`][1], rather than `select`, we'd get back the first result only, which would avoid us needing to call `render_page check_special_file[0]` up on line 20. What do you think?

[1]: https://ruby-doc.org/core-2.6.3/Array.html#method-i-first

Reply:
Agreed, used `select` then `first` to return one item only